jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/quicknewjob/NewJobPresenterTest.java

```diff
@@ -0,0 +1,226 @@
+
+    @Mock
+    EventSourceMock<NotificationEvent> notificationEvent;
+
+    @Mock
+    EventSourceMock<RequestChangedEvent> requestChangedEvent;
+
+    @Mock
+    private ExecutorService executorServicesMock;
+
+    @Mock
+    private NewJobPresenter.NewJobView view;
+
+    @InjectMocks
+    private NewJobPresenter presenter;
+
+    @Before
+    public void setupMocks() {
+        executorServices = new CallerMock<ExecutorService>(executorServicesMock);
+        presenter.setExecutorService(executorServices);
+        presenter.setNotification(notificationEvent);
+        presenter.setRequestChangedEvent(requestChangedEvent);
+        presenter.openNewJobDialog(serverTemplateId);
+    }
+
+    @Test
+    public void testScheduleSuccessfulJob() {
+        Date dueDate = new Date();
+        Long jobId = Long.valueOf(1);
+        when(executorServicesMock.scheduleRequest(eq(serverTemplateId),
+                                                  eq(JOB_TYPE),
+                                                  eq(dueDate),
+                                                  any())).thenReturn(jobId);
+        presenter.createJob(JOB_NAME,
+                            dueDate,
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList<RequestParameterSummary>());
+
+        verify(requestChangedEvent).fire(any(RequestChangedEvent.class));
+        verify(view).hide();
+
+        final ArgumentCaptor<NotificationEvent> captor = ArgumentCaptor.forClass(NotificationEvent.class);
+        verify(notificationEvent).fire(captor.capture());
+        assertEquals(1,
+                     captor.getAllValues().size());
+        assertEquals(NotificationEvent.NotificationType.DEFAULT,
+                     captor.getValue().getType());
+        assertEquals(Constants.INSTANCE.RequestScheduled(jobId),
+                     captor.getValue().getNotification());
+    }
+
+    @Test
+    public void testScheduleRequestParameters() {
+        String param_key = "param_key";
+        String param_value = "param_value";
+        Date dueOnDate = new Date();
+        doAnswer(new Answer() {
+            @Override
+            public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
+                assertTrue(invocationOnMock.getArguments()[0].equals(serverTemplateId));
+                assertTrue(invocationOnMock.getArguments()[1].equals(JOB_TYPE));
+                assertTrue(invocationOnMock.getArguments()[2].equals(dueOnDate));
+                final HashMap<String, String> ctxValues = (HashMap) invocationOnMock.getArguments()[3];
+                assertTrue(ctxValues.get(RequestDataSetConstants.COLUMN_BUSINESSKEY).equals(JOB_NAME));
+                assertTrue(ctxValues.get(RequestDataSetConstants.COLUMN_RETRIES).equals(String.valueOf(JOB_RETRIES)));
+                assertTrue(ctxValues.get(param_key).equals(String.valueOf(param_value)));
+                return null;
+            }
```

Comment:
The arguments are correct now, but the main problem is still there. Assertions in ```answer(...)``` method don't work.
Try to replace
```assertTrue(invocationOnMock.getArguments()[0].equals(serverTemplateId));``` 
with 
```assertTrue(invocationOnMock.getArguments()[0].equals("Fail test"));``` 
and the test will still pass.
Furthermore, the code in ```answer(...)``` method after the first assert statement is not executed at all.
I don't know the exact reason for this behavior, as it comes from Mockito internals obviously, but my hunch is that it has to do with using _doAnswer()_ for stubbing a non-void method.